### PR TITLE
Move get_timestamps utility to sh-py

### PR DESCRIPTION
**Pull request by mlubej**
Changes:
- port the utility function and update it a bit
- add tests

Added to Catalog API because it basically relies upon it, so I see it as an extra utility built on top of the class.

### CODE REVIEW

File: sentinelhub/api/catalog.py [hunk 1]
```diff
@@ -265,3 +268,47 @@ def get_ids(self) -> List[str]:
...
+    :return: A list of timestamps of available observations.
+    """
+    query_filter = None
+    time_difference = time_difference if time_difference is not None else dt.timedelta(seconds=-1)
+    fields = {"include": ["properties.datetime"], "exclude": []}
+
+    if maxcc is not None and data_collection.has_cloud_coverage:
+        if np.issubdtype(type(maxcc), np.integer):
+            raise ValueError('Maximum cloud coverage "maxcc" parameter should be a float on an interval [0, 1]')
+        query_filter = f"eo:cloud_cover < {int(maxcc * 100)}"
```
Comment (mlubej) [inline sentinelhub/api/catalog.py:302]: Not quite sure about raising this error, are the types not enough in this case? is it better to notify the user explicitly to make sure they're using the parameter the right way?

Comment (zigaLuksic) [reply sentinelhub/api/catalog.py:302]: the types don't cover the [0, 1] info. Someone might try to put in percentages.

Comment (mlubej) [reply sentinelhub/api/catalog.py]: constrained the check back to the original.

File: sentinelhub/api/catalog.py [hunk 2]
```diff
@@ -265,3 +266,47 @@ def get_ids(self) -> List[str]:
...
+
+    if maxcc is not None and data_collection.has_cloud_coverage:
+        if isinstance(maxcc, (int, float)) and (maxcc < 0 or maxcc > 1):
+            raise ValueError('Maximum cloud coverage "maxcc" parameter should be a float on an interval [0, 1]')
+        query_filter = f"eo:cloud_cover < {int(maxcc * 100)}"
+
+    if data_collection.service_url is not None:
+        config = config.copy() if config else SHConfig()
+        config.sh_base_url = data_collection.service_url
+
```
Comment (zigaLuksic) [inline sentinelhub/api/catalog.py:305]: this is just so ugly :( but i guess we leave it for now

**Review by zigaLuksic — APPROVED** [review]

File: sentinelhub/api/catalog.py (hunk 2 above)
Comment (mlubej) [reply sentinelhub/api/catalog.py:305]: does anything change with the new config version? I guess I could put the url in the config init, but it doesn't change much.

I wonder what would be the proper way to go about such things.

Comment (zigaLuksic) [reply sentinelhub/api/catalog.py:305]: signatures would need to be changed, 4.0 perhaps

**Pull request merged**